clang/test/Driver/sycl-offload-header-check.cpp

```diff
@@ -0,0 +1,11 @@
+// REQUIRES: system-windows
+// Test the integrated header file name generation.  Name should match the
+// actual path name and not the environment variable setting
+// RUN: mkdir -p %t_DiRnAmE
+// invoke the compiler overriding output temp location
+// RUN: env TMPDIR=%t_dirname TEMP=%t_dirname TMP=%t_dirname  \
+// RUN: %clang_cl -### -fsycl %s 2>&1 | \
+// RUN: FileCheck --check-prefix=CHECK-HEADER %s
+// CHECK-HEADER: clang{{.*}} "-fsycl-int-header=[[HEADER:.+\.h]]"
+// CHECK-HEADER-NOT: clang{{.*}} "-include" "[[HEADER]]"
+// CHECK-HEADER: clang{{.*}} "-include" "{{.*}}_DiRnAmE{{.+}}.h"
```

Comment:
I may be missing something, but is the following implied?
```suggestion
// CHECK-HEADER: clang{{.*}} "-include" "{{.*}}_DiRnAmE\[[HEADER]]"
```
If this is the intended behaviour, than I'd suggest going with this CHECK to add more precision

Reply:
The contents of [[HEADER]] is the full path plus header name.  I just want to verify that the name being passed is the actual path name and not the one as set by the environment variable (_dirname)